dspy/clients/base_lm.py

```diff
@@ -5,6 +5,7 @@
 from dspy.dsp.utils import settings
 from dspy.utils.callback import with_callbacks
 
+MAX_HISTORY_SIZE = 100
```

Comment:
Shall we make this configurable for users? 100 is usually sufficient for me, but there is one time that I need to maintain a long history for trajectory lookup when building `dspy.Refine`.

Reply:
I agree, but will merge now so we don't accumulate too many PRs. Making this configurable is a P2 or P3...

Reply:
I also increased the default to 10_000 so that request (configurable count) is much less urgent now